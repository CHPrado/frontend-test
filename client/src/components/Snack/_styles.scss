$success-color: #4caf50;
$info-color: #2196f3;
$warn-color: #ff9800;
$error-color: #f44336;

.snack-container {
  position: fixed;
  align-items: center;
  justify-content: center;

  bottom: 30px;
  width: 100%;
  z-index: 50;

  &.show {
    display: flex;
  }

  &.hide {
    display: none;
  }
}

.snack-box {
  @extend .flex;

  border-radius: 8px;
  padding: 10px;
  box-shadow: 2px 2px 2px 1px rgba(0, 0, 0, 0.2);

  svg {
    color: #FFF;
  }

  .snack-icon {
    @extend .flex;

    padding: 2px;
    margin: 2px;
  }

  .snack-message {
    width: 100%;
    margin: 10px;

    strong {
      font-size: 15px;
      font-weight: 500;
      color: #FFF;
    }
  }

  .snack-close {
    @extend .flex;
    padding: 2px;
    margin: 2px;
    border-radius: 100%;
    transition: filter 0.2s;
    cursor: pointer;

    &.success {
      background: $success-color;
    }

    &.info {
      background: $info-color;
    }

    &.warn {
      background: $warn-color;
    }

    &.error {
      background: $error-color;
    }

    &:hover {
      filter: brightness(90%);
    }
  }

  &.success {
    background: $success-color;
  }

  &.info {
    background: $info-color;
  }

  &.warn {
    background: $warn-color;
  }

  &.error {
    background: $error-color;
  }
}
