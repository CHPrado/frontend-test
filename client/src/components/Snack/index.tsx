import React from 'react';

import {
  MdCheck, MdErrorOutline, MdInfo, MdWarning, MdClose,
} from 'react-icons/md';

export interface SnackProps {
  type: 'info' | 'warn' | 'success' | 'error' | '';
  message: string;
  open: boolean;
  close: () => void;
}

const Snack: React.FC<SnackProps> = ({
  type = 'info', message, open, close,
}) => {
  const hidden = open ? 'show' : 'hide';
  const icon = (type === 'info' && (<MdInfo />))
                   || (type === 'warn' && (<MdWarning />))
                   || (type === 'success' && (<MdCheck />))
                   || (type === 'error' && (<MdErrorOutline />));

  const handleClose = () => {
    close();
  };

  return (
    <div className={`snack-container ${hidden}`}>
      <div className={`snack-box ${type}`}>
        <div className="snack-icon">
          {icon}
        </div>
        <div className="snack-message">
          <strong>{message}</strong>
        </div>
        <div
          className={`snack-close ${type}`}
          role="button"
          onClick={handleClose}
        >
          <MdClose />
        </div>
      </div>
    </div>
  );
};

export default Snack;