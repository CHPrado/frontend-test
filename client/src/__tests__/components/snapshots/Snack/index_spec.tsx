import React from 'react';
import renderer from 'react-test-renderer';

import Snack from '../../../../components/Snack';

test('it matches the snapshot', () => {
  const tree = renderer.create(
    <Snack 
      type='success'
      message='Test Message'
      open={true}
      close={() => {}}
    />
  ).toJSON()

  expect(tree).toMatchSnapshot();
});
