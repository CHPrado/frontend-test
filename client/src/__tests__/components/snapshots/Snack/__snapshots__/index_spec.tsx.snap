// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 1`] = `
<div
  className="snack-container show"
>
  <div
    className="snack-box success"
  >
    <div
      className="snack-icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
        />
      </svg>
    </div>
    <div
      className="snack-message"
    >
      <strong>
        Test Message
      </strong>
    </div>
    <div
      className="snack-close success"
      onClick={[Function]}
      role="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
  </div>
</div>
`;
